Storage/FirebaseStorageUI/SDWebImageFIRStorageLoader.m

```diff
@@ -0,0 +1,95 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "SDWebImageFIRStorageLoader.h"
+
+@implementation SDWebImageFIRStorageLoader
+
++ (SDWebImageFIRStorageLoader *)sharedLoader {
+  static dispatch_once_t onceToken;
+  static SDWebImageFIRStorageLoader *loader;
+  dispatch_once(&onceToken, ^{
+    loader = [[SDWebImageFIRStorageLoader alloc] init];
+  });
+  return loader;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.defaultMaxImageSize = 10e6;
+  }
+  return self;
+}
+
+#pragma mark - SDImageLoader Protocol
+
+- (BOOL)canLoadWithURL:(NSURL *)url {
+  return url.sd_storageReference;
+}
+
+- (id<SDWebImageOperation>)loadImageWithURL:(NSURL *)url options:(SDWebImageOptions)options context:(SDWebImageContext *)context progress:(SDImageLoaderProgressBlock)progressBlock completed:(SDImageLoaderCompletedBlock)completedBlock {
+  FIRStorageReference *storageRef = url.sd_storageReference;
+  if (!storageRef) {
+    if (completedBlock) {
+      NSError *error = [NSError errorWithDomain:SDWebImageErrorDomain code:SDWebImageErrorInvalidURL userInfo:@{NSLocalizedDescriptionKey : @"Image url is Firebae Storage Reference"}];
```

Comment:
Consider rephrasing the error description: `The provided image url must have an associated FIRStorageReference.`

Reply:
Oops. Typo mistake.